packages/instantsearch.js/src/middlewares/createInsightsMiddleware.ts

```diff
@@ -33,32 +42,55 @@ export type InsightsProps<
   onEvent?: (event: InsightsEvent, insightsClient: TInsightsClient) => void;
 };
 
+const ALGOLIA_INSIGHTS_SRC =
+  'https://cdn.jsdelivr.net/npm/search-insights@2.3.0/dist/search-insights.min.js';
+
 export type CreateInsightsMiddleware = typeof createInsightsMiddleware;
 
 export function createInsightsMiddleware<
   TInsightsClient extends null | InsightsClient
->(props: InsightsProps<TInsightsClient>): InternalMiddleware {
+>(props: InsightsProps<TInsightsClient> = {}): InternalMiddleware {
   const {
     insightsClient: _insightsClient,
     insightsInitParams,
     onEvent,
-  } = props || {};
+  } = props;
+
+  let insightsClient: InsightsClient = _insightsClient || noop;
+
   if (_insightsClient !== null && !_insightsClient) {
-    if (__DEV__) {
-      throw new Error(
-        "The `insightsClient` option is required if you want userToken to be automatically set in search calls. If you don't want this behaviour, set it to `null`."
-      );
-    } else {
-      throw new Error(
-        'The `insightsClient` option is required. To disable, set it to `null`.'
-      );
-    }
+    safelyRunOnBrowser(({ window }: { window: any }) => {
+      const pointer = window.AlgoliaAnalyticsObject || 'aa';
+
+      if (typeof pointer === 'string') {
+        insightsClient = window[pointer];
+      }
+
+      if (!insightsClient) {
+        window.AlgoliaAnalyticsObject = pointer;
+        window[pointer] =
+          window[pointer] ||
+          function (...args: any[]) {
+            (window[pointer].queue = window[pointer].queue || []).push(args);
+          };
+
+        insightsClient = window[pointer];
+
+        try {
+          const script = document.createElement('script');
+          script.async = true;
+          script.src = ALGOLIA_INSIGHTS_SRC;
+          script.onerror = () => {
+            // @TODO: it would be useful to track errors
```

Comment:
Going back to a comment made during discovery, would it be interesting if we add an Insights (Error) (or similar) string in the Algolia user agent when an error occurs?

Reply:
Could be interesting, but it adds something to the search client, but it technically has nothing to do with it

Reply:
That's true, I'll resolve this in favor of [this comment](https://github.com/algolia/instantsearch/pull/5484#discussion_r1112121564).